src/enochecker/enochecker.py

```diff
@@ -265,35 +144,22 @@ def __init__(
         self.requests = requests
 
         self.time_started_at: datetime.datetime = datetime.datetime.now()
-        self.run_id: Optional[int] = run_id
+        self.task_id: int = task.task_id
         self.json_logging: bool = json_logging
 
-        self.method: Optional[str] = method
-        if not address:
-            raise TypeError("must specify address")
-        self.address: str = address
-        if team:
-            warnings.warn(
-                "Passing team as argument to BaseChecker is deprecated, use team_name instead",
-                DeprecationWarning,
-            )
-            team_name = team_name or team
-        self.team: Optional[str] = team_name
-        self.team_id: int = team_id or -1  # TODO: make required
-        if round:
-            warnings.warn(
-                "Passing round as argument to BaseChecker is deprecated, use round_id instead",
-                DeprecationWarning,
-            )
-            round_id = round_id or round
-        self.round: Optional[int] = round_id
-        self.flag_round: Optional[int] = flag_round
-        self.round_length: int = round_length
-        self.flag: Optional[str] = flag
-        if timeout:
-            self.timeout: Optional[int] = timeout // 1000
+        self.method: CheckerMethod = task.method
+        self.address: str = task.address
+        self.team_id: int = task.team_id
+        self.team_name: str = task.team_name
+        self.current_round_id: int = task.current_round_id
+        self.related_round_id: int = task.related_round_id
+        self.flag: Optional[str] = task.flag
+        self.variant_id: int = task.variant_id
+        self.timeout: float = task.timeout / 1000
```

Comment:
we should probably divide by `1000.0` since python assumes integer division on `int / int`